### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by lvwerra**
Adds a contributing guide to the repository.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+Before writing code, we strongly advise you to search through the existing PRs or
+issues to make sure that nobody is already working on the same thing. If you are
+unsure, it is always a good idea to open an issue to get some feedback.
+
+You will need basic `git` proficiency to be able to contribute to
+BigCode. `git` is not the easiest tool to use but it has the greatest
+manual. Type `git --help` in a shell and enjoy. If you prefer books, [Pro
+Git](https://git-scm.com/book/en/v2) is a very good reference.
+
+Follow these steps to start contributing ([supported Python versions](https://github.com/huggingface/transformers/blob/main/setup.py#L426)):
```
Comment (loubnabnl) [inline CONTRIBUTING.md:81]:
```suggestion
Follow these steps to start contributing :
```

**Review by loubnabnl — COMMENTED** [review]
LGTM, I just changed some wording specific to transformers

**Review by harm-devries — COMMENTED** [review]
Thanks @lvwerra! Looks good in general. I'm not 100% sure if we need a feature request option for our repos.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+Did not find it? :( So we can act quickly on it, please follow these steps:
+
+* Include your **OS type and version**, the versions of **Python**, **PyTorch** and
+  **Tensorflow** when applicable;
+* A short, self-contained, code snippet that allows us to reproduce the bug in
+  less than 30s;
+* Provide the *full* traceback if an exception is raised.
+
+### Do you want a new feature (that is not a model)?
```
[the remark this answers is not in the document]
Comment (lvwerra) [reply CONTRIBUTING.md]: I left it for now since maybe for the evaluation harness they will at some point and so we can just copy paste it to all repos :)

**Pull request merged**